### ISSUE TITLE
failure on resliver

### ISSUE THREAD

**Issue by clintar** [zfsonlinux/zfs#4299]
I had some problems with a drive in my zpool, and decided to create a snapshot and attempt to do a zfs send. It took a very long time and didn't fully send, and my system had crashed while that was going on. When I brought the system online, I decided to replace the drive and while reslivering, it failed because there were checksum errors. I assume if I had scrubbed before, it would have found those errors caused by the system crash, but anyway the files that had errors were not important, and I deleted them, but there are 2 remaining in my error list. The current zpool status is 

`  pool: clintards  
 state: DEGRADED  
status: One or more devices is currently being resilvered.  The pool will  
        continue to function, possibly in a degraded state.  
action: Wait for the resilver to complete.  
  scan: resilver in progress since Tue Feb  2 11:54:41 2016
    222G scanned out of 1003G at 239M/s, 0h55m to go
    73.6G resilvered, 22.11% done
config:

```
    NAME                                                      STATE     READ WRITE CKSUM
    clintards                                                 DEGRADED     0     0     2
      raidz1-0                                                DEGRADED     0     0     8
        replacing-0                                           DEGRADED     0     0     0
          5332611342296232703                                 UNAVAIL      0     0     0  was /dev/disk/by-id/ata-ST1000DM005_HD103SJ_S246J9EC432652-part1
          ata-WDC_WD2002FAEX-007BA0_WD-WCAY01444149-part1     ONLINE       0     0     0  (resilvering)
        ata-ST1000DM003-1CH162_Z1D5GTBT-part1                 ONLINE       0     0     0
        ata-WDC_WD2003FZEX-00Z4SA0_WD-WCC1P0541357-part1      ONLINE       0     0     0
    logs
      mirror-1                                                ONLINE       0     0     0
        ata-Samsung_SSD_840_PRO_Series_S12HNEAD201527D-part3  ONLINE       0     0     0
        ata-Samsung_SSD_840_PRO_Series_S12HNEAD201516T-part3  ONLINE       0     0     0
    cache
      ata-Samsung_SSD_840_PRO_Series_S12HNEAD201527D-part4    ONLINE       0     0     0
      ata-Samsung_SSD_840_PRO_Series_S12HNEAD201516T-part4    ONLINE       0     0     0
```

errors: Permanent errors have been detected in the following files:

```
    clintards/clintar-home:<0xa9a22b>
    clintards/clintar-home:<0xa9aefb>`
```

I had tried to scrub and clear a few times, and I believe I just rebooted this last time and it started a resliver, but usually it will finish a resliver, but show the old drive as unavailable and the new drive online (and has i/o in zpool iostat -v), but I assume there is no real redundancy as it will want to finish the resliver before everything is kosher. I just couldn't figure out how to get rid of the 2 files that had errors. ryao suggested i create this issue and determine if the files showing are in the snapshot, which makes sense. I will try to destroy the snapshot and see if everything is better after that.

**Comment by clintar** [comment]
Here is what it looks like after trying to resliver
`  pool: clintards
 state: DEGRADED
status: One or more devices has experienced an error resulting in data
        corruption.  Applications may be affected.
action: Restore the file in question if possible.  Otherwise restore the
        entire pool from backup.
   see: http://zfsonlinux.org/msg/ZFS-8000-8A
  scan: resilvered 326G in 1h12m with 4 errors on Tue Feb  2 13:07:04 2016
config:

        NAME                                                      STATE     READ WRITE CKSUM
        clintards                                                 DEGRADED     0     0    10
          raidz1-0                                                DEGRADED     0     0    40
            replacing-0                                           DEGRADED     0     0     0
              5332611342296232703                                 UNAVAIL      0     0     0  was /dev/disk/by-id/ata-ST1000DM005_HD103SJ_S246J9EC432652-part1
              ata-WDC_WD2002FAEX-007BA0_WD-WCAY01444149-part1     ONLINE       0     0     0
            ata-ST1000DM003-1CH162_Z1D5GTBT-part1                 ONLINE       0     0     0
            ata-WDC_WD2003FZEX-00Z4SA0_WD-WCC1P0541357-part1      ONLINE       0     0     0
        logs
          mirror-1                                                ONLINE       0     0     0
            ata-Samsung_SSD_840_PRO_Series_S12HNEAD201527D-part3  ONLINE       0     0     0
            ata-Samsung_SSD_840_PRO_Series_S12HNEAD201516T-part3  ONLINE       0     0     0
        cache
          ata-Samsung_SSD_840_PRO_Series_S12HNEAD201527D-part4    ONLINE       0     0     0
          ata-Samsung_SSD_840_PRO_Series_S12HNEAD201516T-part4    ONLINE       0     0     0

errors: Permanent errors have been detected in the following files:

        clintards/clintar-home:<0xa9a22b>
        clintards/clintar-home:<0xa9aefb>`

**Comment by clintar** [comment]
Hmm, running a scrub after removing the snapshot still shows errors, and more checksum errors
`  pool: clintards
 state: DEGRADED
status: One or more devices has experienced an error resulting in data
        corruption.  Applications may be affected.
action: Restore the file in question if possible.  Otherwise restore the
        entire pool from backup.
   see: http://zfsonlinux.org/msg/ZFS-8000-8A
  scan: scrub repaired 0 in 0h49m with 2 errors on Tue Feb  2 14:05:32 2016
config:

        NAME                                                      STATE     READ WRITE CKSUM
        clintards                                                 DEGRADED     0     0    14
          raidz1-0                                                DEGRADED     0     0    56
            replacing-0                                           DEGRADED     0     0     0
              5332611342296232703                                 UNAVAIL      0     0     0  was /dev/disk/by-id/ata-ST1000DM005_HD103SJ_S246J9EC432652-part1
              ata-WDC_WD2002FAEX-007BA0_WD-WCAY01444149-part1     ONLINE       0     0     0
            ata-ST1000DM003-1CH162_Z1D5GTBT-part1                 ONLINE       0     0     0
            ata-WDC_WD2003FZEX-00Z4SA0_WD-WCC1P0541357-part1      ONLINE       0     0     0
        logs
          mirror-1                                                ONLINE       0     0     0
            ata-Samsung_SSD_840_PRO_Series_S12HNEAD201527D-part3  ONLINE       0     0     0
            ata-Samsung_SSD_840_PRO_Series_S12HNEAD201516T-part3  ONLINE       0     0     0
        cache
          ata-Samsung_SSD_840_PRO_Series_S12HNEAD201527D-part4    ONLINE       0     0     0
          ata-Samsung_SSD_840_PRO_Series_S12HNEAD201516T-part4    ONLINE       0     0     0

errors: Permanent errors have been detected in the following files:

        clintards/clintar-home:<0xa9a22b>
        clintards/clintar-home:<0xa9aefb>`

**Comment by clintar** [comment]
well, detached old drive, reslivered, and ran a scrub. still have 2 errors
`  pool: clintards
 state: ONLINE
status: One or more devices has experienced an error resulting in data
        corruption.  Applications may be affected.
action: Restore the file in question if possible.  Otherwise restore the
        entire pool from backup.
   see: http://zfsonlinux.org/msg/ZFS-8000-8A
  scan: scrub repaired 0 in 0h49m with 2 errors on Tue Feb  2 16:21:17 2016
config:

        NAME                                                      STATE     READ WRITE CKSUM
        clintards                                                 ONLINE       0     0     8
          raidz1-0                                                ONLINE       0     0    32
            ata-WDC_WD2002FAEX-007BA0_WD-WCAY01444149-part1       ONLINE       0     0     0
            ata-ST1000DM003-1CH162_Z1D5GTBT-part1                 ONLINE       0     0     0
            ata-WDC_WD2003FZEX-00Z4SA0_WD-WCC1P0541357-part1      ONLINE       0     0     0
        logs
          mirror-1                                                ONLINE       0     0     0
            ata-Samsung_SSD_840_PRO_Series_S12HNEAD201527D-part3  ONLINE       0     0     0
            ata-Samsung_SSD_840_PRO_Series_S12HNEAD201516T-part3  ONLINE       0     0     0
        cache
          ata-Samsung_SSD_840_PRO_Series_S12HNEAD201527D-part4    ONLINE       0     0     0
          ata-Samsung_SSD_840_PRO_Series_S12HNEAD201516T-part4    ONLINE       0     0     0

errors: Permanent errors have been detected in the following files:

        clintards/clintar-home:<0xa9a22b>
        clintards/clintar-home:<0xa9aefb>`

**Comment by clintar** [comment]
Any pointers for getting rid of those two error files?

**Comment by clintar** [comment]
Everything I've read where someone had similar errors, there was no way to remove them, and the only thing to be done is set up a new zpool. Seems pretty against what I understood zfs's philosophy to be, but guess that's my only option. You can close this. It didn't really have anything to do with the snapshot.

**Comment by kernelOfTruth** [comment]
@clintar now now - that's a bit radical ;)

you tried running another

``` zpool scrub clintards ```

provided there are no errors this should clear those two "errors"

**Comment by clintar** [comment]
about 20 of them

**Comment by kernelOfTruth** [comment]
@clintar in that case:

``` zpool clear clintards ```

should do it:

```
zpool clear [-F [-n]] pool [device] ...

           Clears device errors in a pool. If no arguments are specified, all device errors within the pool are cleared. If one or more devices is specified, only  those  errors  associated
           with the specified device or devices are cleared.

           -F

               Initiates  recovery  mode  for  an  unopenable pool. Attempts to discard the last few transactions in the pool to return it to an openable state. Not all damaged pools can be
               recovered by using this option. If successful, the data from the discarded transactions is irretrievably lost.

           -n

               Used in combination with the -F flag. Check whether discarding transactions would make the pool openable, but do not actually discard any transactions.
```

**Comment by clintar** [comment]
I've cleared and scrubbed so many times. They just stay, and upon reboot, the drive I added to the pool reslivers again.

**Comment by kernelOfTruth** [comment]
@clintar okay,

you probably meant to look up those specific files by their ID and then delete them off the pool and replace them (or simply forget them),

as far as I know (the last time I looked into this case) there's no way to do a reverse search

CC'ing @behlendorf @dweeezil they surely know

**Comment by dweeezil** [comment]
This issue seems vaguely similar to a problem I've been meaning to look into.  IIRC, the condition is that a crash during a ```zpool detach``` while resilvering is in progress to the same vdev can cause a single vdev to be in a "phantom resilvering" state.

As to the question of converting ```<ds>:0x<objnum>``` back to a path, when either ```zpool status``` or ```zdb -dddd``` don't display the path, it means there aren't any directory entries pointing to it.  A good place to start would be ```zdb -dddd <pool>/<ds> <obj>``` to see what kind of thing it is and what condition it's in.

**Comment by clintar** [comment]
Probably was similar outcome. I've since created a new zpool, moved copied everything over, and created a new pool with the existing drives, filled the entire thing, and did a scrub to make sure there were no issues shown with those, destroyed that and added the drives to the new pool, so can't really check now anyway. Even if I did that, would that have helped me solve anything? Only thing I've seen is that it would be metadata and that there is no way to fix that.
